In [1]:
import math
import pandas as pd

df_sets = (
    pd.read_csv("https://cdn.rebrickable.com/media/downloads/sets.csv.gz")
    .assign(set_num=lambda df: df["set_num"].str[:-2])
    .query("`set_num`.str.isnumeric() & `set_num`.str.len() > 3")
)

set_nums = df_sets["set_num"]

In [2]:
def is_prime(n):
    n = int(n)
    if n < 2 or (n % 2 == 0 and n > 2):
        return False
    return all(n % i for i in range(3, int(math.sqrt(n)) + 1, 2))

def divisors(n):
    n = int(n)
    return [i for i in range(1, n) if n % i == 0]

In [3]:
for i in set_nums:
    if (
        all([is_prime(i[idx]) for idx in range(3)])
        and is_prime(i[0:2])
        and is_prime(i[1:3])
        and is_prime(i[:3])
        and i[-1] != "0"
        and float(int(i[0]) * int(i[1]) * int(i[2])) % int(i[-1]) == 0
    ):
        print(i)

3731
3733
3737
3739


Dies sind also alle Setnummern, welche die erste, mathematische Bedingung erfüllen.
Dann geht es mit dem nächsten Hinweis weiter, der nun händische Arbeit, Recherche und/oder Erfahrung voraussetzt:

> Dann erschien ein größeres Set für Erwachsene, das dennoch stark an dieses kleine Set erinnerte. 

Das einzige sinnvolle Set unter den 4 Nummern scheint hier der [3739 (Blacksmith Shop)](https://amzn.to/3MIDvRi)[<sup>A</sup>]([[Affiliate]]) zu sein, denn vor einigen Jahren erschien ein großes Set für Erwachsene, welches an jenes erinnert, bzw. im Englischen sogar den nahezu gleichen Namen hat ("Blacksmith"): [21325 (Mittelalterliche Schmiede)*](https://amzn.to/3MKgsWo)

> Ein Teiler von dessen Setnummer wiederum ergab [...]

Nun müssen also zunächst alle Teiler dieser Setnummer bestimmt werden.
Hierfür bedienen wir uns einer zuvor definierten Funktion, welche uns eine Liste eben jener Teiler erzeugt:

In [4]:
n = 21325
divisors(n)

[1, 5, 25, 853, 4265]

Basierend auf dieser Liste können wir den nächsten Hinweis bearbeiten:

> [...] wenn man noch eine Ziffer anhängte, ein anderes, recht neues kleines Set [...]":

Der Verweis auf ein "recht neues kleines Set" deutet auf eine 5-stellige Nummer hin, also kann es hier nur mit der 4-stelligen Zahl *4265* weitergehen.
Aus der Liste aller LEGO-Sets suchen wir nun also jene heraus, welche mit der Zeichenfolge *4265* beginnen:

In [5]:
denumerator = '4265'

df_sets.query("`set_num`.str.startswith(@denumerator)")

,set_num,name,year,theme_id,num_parts,img_url
8220,42650,Pet Accessories Shop,2025,494,375,https://cdn.rebrickable.com/media/sets/42650-1...
8221,42651,Horse and Pet Vet Clinic,2025,494,437,https://cdn.rebrickable.com/media/sets/42651-1...
8222,42652,Friendship Tree House Hangout,2025,494,701,https://cdn.rebrickable.com/media/sets/42652-1...
8223,42653,Music Store & Apartment,2025,494,483,https://cdn.rebrickable.com/media/sets/42653-1...
8224,42654,Pony Ranch & Stable,2025,494,742,https://cdn.rebrickable.com/media/sets/42654-1...
8225,42655,Restaurant and Cooking School,2025,494,896,https://cdn.rebrickable.com/media/sets/42655-1...
8226,42656,Heartlake City Airport and Airplane,2025,494,958,https://cdn.rebrickable.com/media/sets/42656-1...
8227,42658,Unicorn & Flamingo Pool Party,2025,494,99,https://cdn.rebrickable.com/media/sets/42658-1...
8228,42659,Friendship Road-Trip Travel Car,2025,494,220,https://cdn.rebrickable.com/media/sets/42659-1...


> [...] das auch schon ein „großes Geschwister“ für Erwachsene hatte, aus derselben Reihe wie das andere.

Hier ergibt sich zunächst die Frage, worauf sich 'das andere' konkret bezieht? 
Nach mehrmaligem Lesen des Textes handelt es sich hierbei aller Wahrscheinlichkeit nach um das zuvor referenzierte "Set für Erwachsene", also die Schmiede.
Diese ist im Rahmen der Ideas-Reihe erschienen.

Folglich wird nun ein Set gesucht, das ebenfalls in der Ideas-Reihe erschienen ist, sich an Erwachsene richtet und mit einem der Sets aus der Liste thematisch übereinstimmt.

Nach Überfliegen der Set-Namen fällt hierbei sofort das [42652 (Friends) Baumhaus*](https://amzn.to/3KIEv7s) auf.
Hierzu gab es ein sehr bekanntes Pendant, dessen Produktionszeitraum aufgrund seiner Beliebtheit sogar [verlängert wurde](https://www.stonewars.de/news/lego-ideas-21318-baumhaus-eol-datum-verschoben/): Das [21318 Baumhaus*](https://amzn.to/4s6vl5q).

Damit können wir nun weiter arbeiten und uns dem nächsten Hinweis widmen:

> Dieses große Geschwister hatte einen Teiler [...]

Somit gilt es nun zunächst alle Teiler der Setnummer *21318* zu bestimmen, wofür wir die gleiche Funktion, wie zuvor, benutzen:

In [6]:
n = 21318
divisors(n)

[1,
 2,
 3,
 6,
 11,
 17,
 19,
 22,
 33,
 34,
 38,
 51,
 57,
 66,
 102,
 114,
 187,
 209,
 323,
 374,
 418,
 561,
 627,
 646,
 969,
 1122,
 1254,
 1938,
 3553,
 7106,
 10659]

 Mit dieser Liste gehen wir weiter:

 > [...] bei dem man nur die Ziffern zu vertauschen brauchte, und schon ergab sich die Setnummer eines Sets, das 2025 EOL geht.

 Wenn das gesuchte Set Ende 2025 vom Markt verschwinden soll, muss es sich um ein aktuelles Set mit 5-stelliger Nummer handeln.
 Diese kann sich also lediglich aus dem Teiler "10659" ergeben.

 Um mögliche Kandidaten zu identifizieren, prüfen wir, welche Sets aus den 5 Ziffern der Zahl *10659* bestehen, indem wir die Ziffern jeweils sortieren und mit der Liste der gesuchten Ziffern vergleichen:

In [9]:
n = sorted("10659")
candidate_sets = [set_num for set_num in set_nums if sorted(set_num) == n]
candidate_sets

['10569',
 '10596',
 '10659',
 '10659',
 '10695',
 '10956',
 '10965',
 '15096',
 '60159',
 '60195',
 '65109']

Um herauszufinden, welches dieser Sets am Ende des Jahres 2025 EOL geht, sortieren wir zunächst die Liste nach der Spalte "year".
Dann können wir beginnend mit dem neuesten Set prüfen, welche hiervon auf der Stonewars [EOL-Liste 2025](https://www.stonewars.de/eol-2025/) stehen:

In [10]:
df_sets.query("`set_num` in @candidate_sets").sort_values("year")

,set_num,name,year,theme_id,num_parts,img_url
14578,65109,Pahrak Nuhvok Gahlok Co-Pack,2002,324,0,https://cdn.rebrickable.com/media/sets/65109-1...
765,10659,Blue Suitcase,2013,35,152,https://cdn.rebrickable.com/media/sets/10659-1...
766,10659,Vehicle Suitcase,2014,591,152,https://cdn.rebrickable.com/media/sets/10659-2...
710,10569,Treasure Attack,2014,647,46,https://cdn.rebrickable.com/media/sets/10569-1...
800,10695,Creative Building Box,2015,621,580,https://cdn.rebrickable.com/media/sets/10695-1...
1823,15096,Ninjago Rebooted Easy Set,2015,777,0,https://cdn.rebrickable.com/media/sets/15096-1...
736,10596,Disney Princess Collection,2015,640,63,https://cdn.rebrickable.com/media/sets/10596-1...
13477,60159,Jungle Halftrack Mission,2017,614,378,https://cdn.rebrickable.com/media/sets/60159-1...
13519,60195,Arctic Mobile Exploration Base,2018,65,786,https://cdn.rebrickable.com/media/sets/60195-1...
1058,10956,Amusement Park,2021,632,95,https://cdn.rebrickable.com/media/sets/10956-1...


Entsprechend dieser Prüfung, geht die [**10965 Schwimmender Tierzug***](https://amzn.to/4972Cos) EOL!

Damit haben wir das gesuchte Set und die Lösung für das heutige Rätsel gefunden :-)